pkg/goalert/service_test.go

```diff
@@ -78,148 +78,289 @@ func Test_NewRequest(t *testing.T) {
 
 func Test_CreateService(t *testing.T) {
 
-	mockRespData := []byte(`{"data": {"createService": {"id": "456"}}}`)
-
-	// Test successful response
-	data := &Data{
-		Name:               "Test",
-		Description:        "Test service",
-		Favorite:           false,
-		EscalationPolicyID: "123",
-	}
-
-	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		resp := mockRespData
-
-		w.Header().Set("Content-Type", "application/json")
-		w.WriteHeader(http.StatusOK)
-		if _, err := w.Write(resp); err != nil {
-			t.Fatalf("Unexpected error writing response from httptest server")
-		}
-	}))
-	defer mockServer.Close()
-
-	t.Setenv(config.GoalertApiEndpointEnvVar, mockServer.URL)
-
-	mockClient := &GraphqlClient{
-		sessionCookie: &http.Cookie{
-			Name: "test_cookie",
+	tests := []struct {
+		name        string
+		data        *Data
+		expectedID  string
+		respData    []byte
+		expectedErr bool
+	}{
+		{
+			name: "Succesful createService",
+			data: &Data{
+				Name:               "Test",
+				Description:        "Test service",
+				Favorite:           false,
+				EscalationPolicyID: "123",
+			},
+			expectedID:  "456",
+			respData:    []byte(`{"data": {"createService": {"id": "456"}}}`),
+			expectedErr: false,
+		},
+		{
+			name: "Unsuccesful createService",
+			data: &Data{
+				Name:               "Test2",
+				Description:        "Test service",
+				Favorite:           false,
+				EscalationPolicyID: "123-bad",
+			},
+			expectedID:  "",
+			respData:    []byte(`{"data":{"createService":null}}`),
+			expectedErr: false,
+		},
+		{
+			name: "Failed unmarshaling response",
+			data: &Data{
+				Name:               "Test3",
+				Description:        "Test service",
+				Favorite:           false,
+				EscalationPolicyID: "890",
+			},
+			expectedID:  "",
+			respData:    []byte(`nmuyrufcewrqrew`),
+			expectedErr: true,
 		},
-		httpClient: mockServer.Client(),
 	}
-	expectedID := "456"
-	actualID, err := mockClient.CreateService(data)
-	assert.NoError(t, err)
-	assert.Equal(t, expectedID, actualID)
-}
-
-func Test_CreateIntegrationKey(t *testing.T) {
 
-	// Define expected input data and response data
-	testData := &Data{
-		Id:   "123",
-		Type: "test",
-		Name: "Test Integration Key",
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				resp := test.respData
+
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				if _, err := w.Write(resp); err != nil {
```

Comment:
```suggestion
				if _, err := w.Write(resp); err != nil {
```
Is it necessary to explicitly set the header content type and status? Seems like the following Write does both already.

> 	// Write writes the data to the connection as part of an HTTP reply.
	//
	// If WriteHeader has not yet been called, Write calls
	// WriteHeader(http.StatusOK) before writing the data. If the Header
	// does not contain a Content-Type line, Write adds a Content-Type set
	// to the result of passing the initial 512 bytes of written data to
	// DetectContentType.
> https://pkg.go.dev/net/http#ResponseWriter.Write